fetchNixpkgs.nix

```diff
@@ -0,0 +1,52 @@
+{ rev                             # The Git revision of nixpkgs to fetch
+, sha256                          # The SHA256 of the downloaded data
+, sha256unpacked
+, system ? builtins.currentSystem # This is overridable if necessary
+}:
+
+with {
+  ifThenElse = { bool, thenValue, elseValue }: (
+    if bool then thenValue else elseValue);
+};
```

Comment:
Let's use `let ... in` and `if ... then ... else` instead. I recognize the `ifThenElse` pattern from elsewhere and it's been cited as an anti-pattern.